src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs

```diff
@@ -733,80 +736,56 @@ private void GenerateResourceLoader(IndentedStringBuilder writer)
 
 			writer.AppendLineIndented($"global::Windows.ApplicationModel.Resources.ResourceLoader.AddLookupAssembly(GetType().Assembly);");
 
-			foreach (var reference in _metadataHelper.Compilation.ExternalReferences)
+			foreach (var reference in _metadataHelper.Compilation.References)
 			{
-				string? GetFilePath()
+                if(_metadataHelper.Compilation.GetAssemblyOrModuleSymbol(reference) is IAssemblySymbol assembly)
+                {
+					BuildResourceLoaderFromAssembly(writer, assembly);
+                }
+                else
+                {
+                    throw new InvalidOperationException($"Unsupported resource type for {reference.Display} ({reference.GetType()})");
+                }
+            }
+		}
+
+        private void BuildResourceLoaderFromAssembly(IndentedStringBuilder writer, IAssemblySymbol assembly)
+        {
+			if (assembly.GetAllAttributes().Any(a =>
+				SymbolEqualityComparer.Default.Equals(a.AttributeClass, _assemblyMetadataSymbol)
+				&& a.ConstructorArguments.Length == 2
+				&& a.ConstructorArguments[0].Value?.ToString() == "UnoHasLocalizationResources"
+				&& (a.ConstructorArguments[1].Value?.ToString().Equals("True", StringComparison.OrdinalIgnoreCase) ?? false)))
+			{
+				if (_isWasm)
 				{
-					if (reference is PortableExecutableReference per && File.Exists(per.FilePath))
-					{
-						return per.FilePath;
-					}
-					else if (File.Exists(reference.Display))
+					var anchorType = assembly.Modules.First().GlobalNamespace.GetNamespaceTypes().FirstOrDefault(s => s.IsLocallyPublic(_metadataHelper.Compilation.Assembly.Modules.First()));
```

Comment:
This might be expensive.

Reply:
it's taking about 1.0ms on `Uno.UI.RemoteControl`, then 0.2ms `Uno.UI.RuntimeTests`. Most of the lookup is don through enumerables provided by `GetNamespaceTypes()`, so until we find a better way to handle this (this is a mono-wasm limitation), we'll need to live with it... it was previously done through a cecil lookup that was not particularly efficient either:

https://github.com/unoplatform/uno/blob/3cb7e5f97a9226ba3d0b53756e61a0d94832169c/src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs#L796